diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/vorbis-tools-1.1.1/ogg123/buffer.c src/vorbis-tools-1.1.1/ogg123/buffer.c
--- origsrc/vorbis-tools-1.1.1/ogg123/buffer.c	2005-06-03 05:15:09.000000000 -0500
+++ src/vorbis-tools-1.1.1/ogg123/buffer.c	2007-07-29 21:52:14.015625000 -0500
@@ -308,6 +308,11 @@
     }
     else {
 
+      if (buf->cancel_flag || sig_request.cancel)
+	{
+	  UNLOCK_MUTEX(buf->mutex);
+	  break;
+	}
       /* No room for more data, wait until there is */
       DEBUG("No room for data in buffer.  Waiting.");
       COND_WAIT(buf->write_cond, buf->mutex);
